core/src/main/scala/org/apache/spark/status/api/v1/ApplicationListResource.scala

```diff
@@ -37,15 +38,23 @@ private[v1] class ApplicationListResource extends ApiRequestContext {
     val includeCompleted = status.isEmpty || status.contains(ApplicationStatus.COMPLETED)
     val includeRunning = status.isEmpty || status.contains(ApplicationStatus.RUNNING)
 
-    uiRoot.getApplicationInfoList.filter { app =>
-      val anyRunning = app.attempts.exists(!_.completed)
-      // if any attempt is still running, we consider the app to also still be running;
-      // keep the app if *any* attempts fall in the right time window
-      ((!anyRunning && includeCompleted) || (anyRunning && includeRunning)) &&
-      app.attempts.exists { attempt =>
-        isAttemptInRange(attempt, minDate, maxDate, minEndDate, maxEndDate, anyRunning)
+    val applicationInfoList = uiRoot.getApplicationInfoList
```

Comment:
Cannot close `KVStoreIterator` in `getApplicationList` because it return an `Iterator`